README.md

M+ FONTS Based `ggplot2` Themes.
--------------------------------

[![Project Status: Active – The project is being actively developed.](http://www.repostatus.org/badges/latest/active.svg)](http://www.repostatus.org/#active) [![Last-changedate](https://img.shields.io/badge/last%20change-2017--02--27-green.svg)](/commits/master) [![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT) [![keybase verified](https://img.shields.io/badge/keybase-verified-brightgreen.svg)](https://gist.github.com/bhaskarvk/46fbf2ba7b5713151d7e)

[![Travis-CI Build Status](https://travis-ci.org/bhaskarvk/fontMPlus.svg?branch=master)](https://travis-ci.org/bhaskarvk/fontMPlus) [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/bhaskarvk/fontMPlus?branch=master&svg=true)](https://ci.appveyor.com/project/bhaskarvk/fontMPlus)

[![minimal R version](https://img.shields.io/badge/R%3E%3D-3.0.0-6666ff.svg)](https://cran.r-project.org/) [![packageversion](https://img.shields.io/badge/Package%20version-0.1.1-orange.svg?style=flat-square)](commits/master) [![CRAN\_Status\_Badge](http://www.r-pkg.org/badges/version/fontMPlus)](https://cran.r-project.org/package=fontMPlus) [![](http://cranlogs.r-pkg.org/badges/grand-total/fontMPlus)](http://cran.rstudio.com/web/packages/fontMPlus/index.html)

------------------------------------------------------------------------

This is an add-on pacakge for [hrbrthemes](https://github.com/hrbrmstr/hrbrthemes) pacakge. It provides seven ggplot2 themes based on [M+ FONTS](http://mplus-fonts.osdn.jp/about-en.html).

The M+ fonts are a font family under the Free license. You can use, copy, and distribute them, with or without modification, either commercially or noncommercially. The font family provides multilingual glyphs. The fonts provide Kana, over 5,000 Kanji, Basic Latin, Latin-1 Supplement, Latin Extended-A, and IPA Extensions glyphs. Most of the Greek, Cyrillic, Vietnamese, and extended glyphs and symbols are included too. So the fonts are in conformity with ISO-8859-1, 2, 3, 4, 5, 7, 9, 10, 13, 14, 15, 16, Windows-1252, T1, and VISCII encoding.

In order to keep the R package size under 5MB, which is a CRAN requirement, we include only the basic Latin glyphs with R package. But you can clone this repo and use complete glyph set as described in the documentation of `import_mplus()` function.

The package is released under MIT License while the M+ fonts are released under a [license](http://mplus-fonts.osdn.jp/about-en.html#license) that permits free usage, distribution, modification both commercially and non-commercially.

### Installation

``` r
install.packages('hrbrthemes')

if(!require(devtools)) {
  install.packages("devtools")
}
devtools::install_github('bhaskarvk/fontMPlus')


# Required only once to import basic latin glpyh fonts included with this package.
fontMPlus::import_mplus()

# To import complete glyph set including Kana/Kanji glyphs use one of the two approaches below.
# NOTE: You don't need to do this if you don't need Kana/Kanji glyphs.

# Download and extract https://osdn.net/projects/mplus-fonts/downloads/62344/mplus-TESTFLIGHT-062.tar.xz/ to some directory.
fontMPlus::import_mplus(font_dir='directory containing M+ fonts')

# OR

# git clone this repo.
# cd fontMPlus (where fontMPlus is the directory where you have cloned this repo)
fontMPlus::import_mplus(font_dir='data-raw/complete-glpyhs')
```

The `fontMPlus::import_mplus()` call will import the M+ fonts in your `extrafont` database. You will then need to install the M+ fonts in the directory mentioned in the output of that call as per your operating system's way of installing fonts.

### Usage

#### M+ C Type 1

``` r
library(ggplot2)
library(fontMPlus)

gg <- ggplot(mtcars, aes(mpg, wt)) +
  geom_point() +
  labs(x="Fuel efficiency (mpg)", y="Weight (tons)",
       title="Seminal ggplot2 scatterplot example",
       subtitle="A plot that is only useful for demonstration purposes",
       caption="Brought to you by the letter 'g'")

gg + theme_ipsum_mplus_c1()
```

<img src="README_figs/README-c1-1.png" width="672" />

#### M+ C Type 2

``` r
gg + theme_ipsum_mplus_c2()
```

<img src="README_figs/README-c2-1.png" width="672" />

#### M+ P Type 1

``` r
gg + theme_ipsum_mplus_p1()
```

<img src="README_figs/README-p1-1.png" width="672" />

#### M+ P Type 2

``` r
gg + theme_ipsum_mplus_p2()
```

<img src="README_figs/README-p2-1.png" width="672" />

#### M+ M Type 1

``` r
gg + theme_ipsum_mplus_m1()
```

<img src="README_figs/README-m1-1.png" width="672" />

#### M+ M Type 2

``` r
gg + theme_ipsum_mplus_m2()
```

<img src="README_figs/README-m2-1.png" width="672" />

#### M+ MN Type 1

``` r
gg + theme_ipsum_mplus_mn1()
```

<img src="README_figs/README-mn1-1.png" width="672" />

### Code of Conduct

Please note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.


NEWS.md
# fontMPlus 0.1.1

## New Features

* Seven ggplot2 themes based on [M+ FONTS](http://mplus-fonts.osdn.jp/about-en.html).





R/mplus.R
#' Import MPlus font for use in hrbrthemes
#'
#' @note This will take care of ensuring PDF/PostScript usage. The location of the
#'   font directory is displayed after the base import is complete. It is highly
#'   recommended that you install them on your system the same way you would any
#'   other font you wish to use in other programs.
#' @param font_dir Location of M+ TTF fonts.
#' This package includes only basic Latin glyph fonts. If you need multilingual glyphs
#' including Kana/Kanji glyphs, download and extract complete M+ fonts from \url{http://mplus-fonts.osdn.jp/about-en.html} and pass the directory name where you have extracted the fonts.
#' @export
import_mplus <- function(font_dir =system.file("fonts", "mplus", package="fontMPlus")) {

  suppressWarnings(suppressMessages(extrafont::font_import(font_dir, prompt=FALSE)))

  message(sprintf("You will likely need to install these fonts on your system as well.
                  You can find them in [%s]", font_dir))

}

#' M+ Font Table
#'
#' fonttable of M+ fonts. Not intended for general use.
#'
#' @format data.frame
"mplus.fonttable"


#' M+ Font Families
#'
#' Convenient List of font families in M+ fonts.
#' You can use this to pass family name options to any one of the seven themes provided by this package.
#'
#' @format list
"mplus.fontfamilies"



R/theme_ipsum_mplus.R
globalVariables(c('mplus.fontfamilies')) # To make R CMD Check happy

#' ggplot2 theme based on M+ C Type1 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_c1}{M+ C Type 1} fonts are a combination of fixed-fullwidth M+ Type-1 for Japanese and proportional M+ C Type-1 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_c1 <- function(
  base_family       = mplus.fontfamilies$`1c`,
  plot_title_family = mplus.fontfamilies$`1c-bold`,
  subtitle_family   = mplus.fontfamilies$`1c-medium`,
  strip_text_family = mplus.fontfamilies$`1c-regular`,
  caption_family    = mplus.fontfamilies$`1c-light`,
  axis_title_family = mplus.fontfamilies$`1c-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ C Type2 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_c2}{M+ C Type 2} fonts are a combination of fixed-fullwidth M+ Type-2 for Japanese and proportional M+ C Type-2 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_c2 <- function(
  base_family       = mplus.fontfamilies$`2c`,
  plot_title_family = mplus.fontfamilies$`2c-bold`,
  subtitle_family   = mplus.fontfamilies$`2c-medium`,
  strip_text_family = mplus.fontfamilies$`2c-regular`,
  caption_family    = mplus.fontfamilies$`2c-light`,
  axis_title_family = mplus.fontfamilies$`2c-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ P Type1 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_p1}{M+ P Type 1} fonts are a combination of fixed-fullwidth M+ Type-1 for Japanese and proportional M+ P Type-1 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_p1 <- function(
  base_family       = mplus.fontfamilies$`1p`,
  plot_title_family = mplus.fontfamilies$`1p-bold`,
  subtitle_family   = mplus.fontfamilies$`1p-medium`,
  strip_text_family = mplus.fontfamilies$`1p-regular`,
  caption_family    = mplus.fontfamilies$`1p-light`,
  axis_title_family = mplus.fontfamilies$`1p-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ P Type2 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_p2}{M+ P Type 2} fonts are a combination of fixed-fullwidth M+ Type-2 for Japanese and proportional M+ P Type-2 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_p2 <- function(
  base_family       = mplus.fontfamilies$`2p`,
  plot_title_family = mplus.fontfamilies$`2p-bold`,
  subtitle_family   = mplus.fontfamilies$`2p-medium`,
  strip_text_family = mplus.fontfamilies$`2p-regular`,
  caption_family    = mplus.fontfamilies$`2p-light`,
  axis_title_family = mplus.fontfamilies$`2p-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ M Type1 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_m1}{M+ M Type 1} fonts are a combination of fixed-fullwidth M+ Type-1 for Japanese and fixed-halfwidth M+ M Type-1 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_m1 <- function(
  base_family       = mplus.fontfamilies$`1m`,
  plot_title_family = mplus.fontfamilies$`1m-bold`,
  subtitle_family   = mplus.fontfamilies$`1m-medium`,
  strip_text_family = mplus.fontfamilies$`1m-regular`,
  caption_family    = mplus.fontfamilies$`1m-light`,
  axis_title_family = mplus.fontfamilies$`1m-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ M Type2 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_m2}{M+ M Type 2} fonts are a combination of fixed-fullwidth M+ Type-2 for Japanese and fixed-halfwidth M+ M Type-2 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_m2 <- function(
  base_family       = mplus.fontfamilies$`2m`,
  plot_title_family = mplus.fontfamilies$`2m-bold`,
  subtitle_family   = mplus.fontfamilies$`2m-medium`,
  strip_text_family = mplus.fontfamilies$`2m-regular`,
  caption_family    = mplus.fontfamilies$`2m-light`,
  axis_title_family = mplus.fontfamilies$`2m-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}

#' ggplot2 theme based on M+ MN Type1 fonts.
#'
#' \href{http://mplus-fonts.osdn.jp/design.html#mplus_mn1}{M+ MN Type 1} fonts are a combination of fixed-fullwidth M+ Type-1 for Japanese and fixed-halfwidth M+ MN Type-1 for alphabets.
#' @inherit hrbrthemes::theme_ipsum
#' @inheritDotParams hrbrthemes::theme_ipsum -base_family -plot_title_family
#'   -subtitle_family -strip_text_family -caption_family -axis_title_family
#' @export
theme_ipsum_mplus_mn1 <- function(
  base_family       = mplus.fontfamilies$`1mn`,
  plot_title_family = mplus.fontfamilies$`1mn-bold`,
  subtitle_family   = mplus.fontfamilies$`1mn-medium`,
  strip_text_family = mplus.fontfamilies$`1mn-regular`,
  caption_family    = mplus.fontfamilies$`1mn-light`,
  axis_title_family = mplus.fontfamilies$`1mn-light`,
  ...) {
  hrbrthemes::theme_ipsum(
    base_family = base_family,
    plot_title_family = plot_title_family,
    subtitle_family = subtitle_family,
    strip_text_family = strip_text_family,
    caption_family = caption_family,
    axis_title_family = axis_title_family,
    ...
  )
}


R/fontMPlus-package.r
#' Additional ggplot2 themes using M+ fonts.
#'
#' This is an add-on pacakge for \code{\link{hrbrthemes}} pacakge.
#'  It provides seven ggplot2 themes based on
#'   \href{http://mplus-fonts.osdn.jp/about-en.html}{M+} fonts.
#'
#' @section M+ FONTS:
#'
#'  The M+ FONTS are a font family under the Free license.
#'  You can use, copy, and distribute them, with or without modification,
#'   either commercially or noncommercially.
#'  The font family provides multilingual glyphs.
#' The fonts provide Kana, over 5,000 Kanji, Basic Latin, Latin-1 Supplement,
#' Latin Extended-A, and IPA Extensions glyphs. Most of the Greek, Cyrillic,
#'  Vietnamese, and extended glyphs and symbols are included too.
#'  So the fonts are in conformity with
#'  ISO-8859-1, 2, 3, 4, 5, 7, 9, 10, 13, 14, 15, 16, Windows-1252, T1, and VISCII encoding.
#'
#' @section One-time setup:
#'
#' Before using this theme you need to call \code{\link{import_mplus}()} which will
#'  import the M+ fonts in your \code{\link{extrafont}} database.
#'  Only basic Latin glyph fonts are provided with this package, but you can download and
#'  install complete glpyh sets including Kana/Kanji glyphs using the `font_dir` argument.
#'  After this you will also need to install the M+ fonts in the directory
#'  mentioned in the output #'  of \code{\link{import_mplus}()} as per your
#'  operating system's way of installing fonts.
#'
#' @name fontMPlus
#' @docType package
#' @import extrafont hrbrthemes
NULL
